pkg/traffic/gateway.go

```diff
@@ -0,0 +1,182 @@
+package traffic
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/utils/strings/slices"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/_internal/slice"
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/apis/v1alpha1"
+	status "github.com/Kuadrant/multi-cluster-traffic-controller/pkg/syncer/status"
+)
+
+func NewGateway(g *gatewayv1beta1.Gateway) Interface {
+	return &Gateway{Gateway: g}
+}
+
+type Gateway struct {
```

Comment:
It looks like we haven't defined an interface for this object? Should we be using one? Are gateways abstracted like this, for the same reason as ingresses (i.e. routes might come in future).

Reply:
I've attempted to use the traffic Interface for the Gateway resource as much as possible.
Is there some wiring that I've missed?

Reply:
I missed the reused traffic interface, that's cool.